Octokit/Models/Request/SearchCommitsRequest.cs

```diff
@@ -0,0 +1,305 @@
+﻿using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Linq;
+using Octokit.Internal;
+
+namespace Octokit
+{
+    /// <summary>
+    /// Searching Code/Files
+    /// https://developer.github.com/v3/search/#search-commits
+    /// </summary>
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    public class SearchCommitsRequest : BaseSearchRequest
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SearchCommitsRequest"/> class.
+        /// </summary>
+        public SearchCommitsRequest() : base()
+        {
+            Repos = new RepositoryCollection();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SearchCommitsRequest"/> class.
+        /// </summary>
+        /// <param name="term">The search term.</param>
+        public SearchCommitsRequest(string term) : base(term)
+        {
+            Repos = new RepositoryCollection();
+        }
+
+        /// <summary>
+        /// Optional Sort field. Can only be indexed, which indicates how recently 
+        /// a file has been indexed by the GitHub search infrastructure. 
+        /// If not provided, results are sorted by best match.
+        /// </summary>
+        /// <remarks>
+        /// http://developer.github.com/v3/search/#search-commits
+        /// </remarks>
+        public CommitSearchSort? SortField { get; set; }
+        public override string Sort
+        {
+            get { return SortField.ToParameter(); }
+        }
+
+        /// <summary>
+        /// Finds commits authored by specified name
+        /// </summary>
+        /// <remarks>
+        /// https://help.github.com/en/github/searching-for-information-on-github/searching-commits#search-by-author-or-committer
+        /// </remarks>
+        public string Author;
```

Comment:
We have a convention in this codebase to ensure response models are properties defined with `{ get; protected set; }`. Are you able to add these?